pkg/sensors/accelerator/device/device.go

```diff
@@ -0,0 +1,156 @@
+package device
+
+import (
+	"sync"
+	"time"
+
+	"golang.org/x/exp/maps"
+	"k8s.io/klog/v2"
+)
+
+const (
+	DUMMY = iota
+	HABANA
+	DCGM
+	NVML
+)
+
+var (
+	globalRegistry *DeviceRegistry
+	once           sync.Once
+)
+
+type DeviceType int
+
+// Function prototype to create a new deviceCollector.
+type deviceStartupFunc func() DeviceInterface
+type DeviceRegistry struct {
+	gpuDevices   map[DeviceType]deviceStartupFunc // Static map of supported gpuDevices.
+	dummyDevices map[DeviceType]deviceStartupFunc // Static map of supported dummyDevices.
```

Comment:
We only use the dummy devices for testing amd is built in to have at least one thing built in to the pkg. I thought it would be more efficient to have separate registries for different device types (one being qat again in the future,  as the code that's there today for qat will not work in RHEL and the plan is to fix that in the very near future)...